### PR TITLE
Rabl engine caching

### PR DESCRIPTION
**Pull request by zenspider**
Hi hi hi...

I just finished up doing a code audit for a client and this is some of the work that came out of it that they wanted pushed upstream.

Let me know if you have any questions or want any changes to the PR. There are a couple optional commits in here but I think they help clarify things a bit... Totally up to you if you want them to go in.

### CODE REVIEW

**Comment by nesquena** [comment]
This is great, thanks!

**Comment by zenspider** [comment]
Damn! Nice response time! Thank you.

Any chance I could nudge you into a release?

**Comment by nesquena** [comment]
Sorry for the delay, 0.14.1 has been pushed, thanks again

File: lib/rabl/helpers.rb [hunk 1]
```diff
@@ -81,8 +83,10 @@ def is_object?(obj, follow_symbols = true)
     # is_collection?([]) => true
     def is_collection?(obj, follow_symbols = true)
       data_obj = follow_symbols ? data_object(obj) : obj
-      data_obj && data_obj.respond_to?(:map) && data_obj.respond_to?(:each) &&
-        !(data_obj.is_a?(Struct) || defined?(Hashie::Mash) && data_obj.is_a?(Hashie::Mash))
+      data_obj &&
+        data_obj.is_a?(Enumerable) &&
```
Comment (nerdrew) [inline lib/rabl/helpers.rb]: This seems to be a breaking change for my app when I upgraded from `0.14.0` to `0.14.1`. I'm not sure if it is related to https://github.com/nesquena/rabl/issues/724.

I have a collection that responds to map, but it is not an enumerable.

Comment (nerdrew) [reply lib/rabl/helpers.rb]: Do ActiveRecord relations include Enumerable? It looks like they do not in rails 4.2.

```
> Widget.all.class
=> Widget::ActiveRecord_Relation
> Widget.all < Enumerable
=> nil
> Widget.all.respond_to?(:each)
=> true
> Widget.all.respond_to?(:map)
=> true
```

**Pull request merged**